community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/commands/predicates/Predicate.scala

```diff
@@ -152,6 +152,7 @@ case class PropertyExists(identifier: Expression, propertyKey: KeyToken) extends
   def isMatch(m: ExecutionContext)(implicit state: QueryState): Option[Boolean] = identifier(m) match {
     case pc: Node         => Some(propertyKey.getOptId(state.query).exists(state.query.nodeOps.hasProperty(pc.getId, _)))
     case pc: Relationship => Some(propertyKey.getOptId(state.query).exists(state.query.relationshipOps.hasProperty(pc.getId, _)))
+    case IsMap(map)       => Some(map(state.query).contains(propertyKey.name))
```

Comment:
but in this case you want to treat nodes and relationships separately, e.g. do `state.query.nodeOps.hasProperty(pc.getId, _)` for nodes